This patch fixes the behavior of ORC_PREFER_STATIC_ZLIB, and was corrected in 1.7.5:
https://github.com/apache/orc/commit/1afc31d6c04729d7e194a6423c690af4519aab33

--- cmake_modules/ThirdpartyToolchain.cmake	2022-04-09 14:50:16.000000000 -0700
+++ cmake_modules/zlibtest.cmake	2023-02-14 11:44:28.000000000 -0700
@@ -145,9 +145,9 @@
 add_library (orc_zlib INTERFACE)
 add_library (orc::zlib ALIAS orc_zlib)
 if (ORC_PREFER_STATIC_ZLIB AND ${ZLIB_STATIC_LIB})
-  target_link_libraries (orc_zlib INTERFACE ${ZLIB_LIBRARY})
-else ()
   target_link_libraries (orc_zlib INTERFACE ${ZLIB_STATIC_LIB})
+else ()
+  target_link_libraries (orc_zlib INTERFACE ${ZLIB_LIBRARY})
 endif ()
 target_include_directories (orc_zlib SYSTEM INTERFACE ${ZLIB_INCLUDE_DIR})
